diff --git a/rust/sedona-raster-functions/Cargo.toml b/rust/sedona-raster-functions/Cargo.toml
index 36ab816..e70b35f 100644
--- a/rust/sedona-raster-functions/Cargo.toml
+++ b/rust/sedona-raster-functions/Cargo.toml
@@ -30,6 +30,10 @@ rust-version.workspace = true
 [lints.clippy]
 result_large_err = "allow"
 
+[features]
+default = ["spatial-predicates"]
+spatial-predicates = ["dep:sedona-tg"]
+
 [dependencies]
 arrow-schema = { workspace = true }
 arrow-array = { workspace = true }
@@ -41,7 +45,7 @@ sedona-expr = { workspace = true }
 sedona-geometry = { workspace = true }
 sedona-raster = { workspace = true }
 sedona-schema = { workspace = true }
-sedona-tg = { workspace = true }
+sedona-tg = { workspace = true, optional = true }
 sedona-proj = { workspace = true }
 serde_json = { workspace = true }
 wkb = { workspace = true }
diff --git a/rust/sedona-raster-functions/src/lib.rs b/rust/sedona-raster-functions/src/lib.rs
index 592a0b4..cbbc4ce 100644
--- a/rust/sedona-raster-functions/src/lib.rs
+++ b/rust/sedona-raster-functions/src/lib.rs
@@ -30,6 +30,7 @@ pub mod rs_pixel_functions;
 pub mod rs_rastercoordinate;
 pub mod rs_setsrid;
 pub mod rs_size;
+#[cfg(feature = "spatial-predicates")]
 pub mod rs_spatial_predicates;
 pub mod rs_srid;
 pub mod rs_worldcoordinate;
diff --git a/rust/sedona-raster-functions/src/register.rs b/rust/sedona-raster-functions/src/register.rs
index e86fe8e..32804d9 100644
--- a/rust/sedona-raster-functions/src/register.rs
+++ b/rust/sedona-raster-functions/src/register.rs
@@ -65,9 +65,6 @@ pub fn default_function_set() -> FunctionSet {
         crate::rs_setsrid::rs_set_srid_udf,
         crate::rs_srid::rs_crs_udf,
         crate::rs_srid::rs_srid_udf,
-        crate::rs_spatial_predicates::rs_contains_udf,
-        crate::rs_spatial_predicates::rs_intersects_udf,
-        crate::rs_spatial_predicates::rs_within_udf,
         crate::rs_worldcoordinate::rs_rastertoworldcoord_udf,
         crate::rs_worldcoordinate::rs_rastertoworldcoordx_udf,
         crate::rs_worldcoordinate::rs_rastertoworldcoordy_udf,
@@ -75,5 +72,13 @@ pub fn default_function_set() -> FunctionSet {
 
     register_aggregate_udfs!(function_set,);
 
+    #[cfg(feature = "spatial-predicates")]
+    register_scalar_udfs!(
+        function_set,
+        crate::rs_spatial_predicates::rs_contains_udf,
+        crate::rs_spatial_predicates::rs_intersects_udf,
+        crate::rs_spatial_predicates::rs_within_udf,
+    );
+
     function_set
 }
